store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedGraphStorage.java

```diff
@@ -163,21 +169,25 @@ public Collection<GraphSerialisable> getAll(final User user) {
      *
      * @param graphId the graphId to remove.
      * @param user    to match visibility against.
+     * @param removeCache to remove associated caches with this graph.
      * @return if a graph was removed.
      * @see #isValidToView(User, FederatedAccess)
      */
-    public boolean remove(final String graphId, final User user) {
-        return remove(graphId, federatedAccess -> federatedAccess.hasWriteAccess(user));
+    public boolean remove(final String graphId, final User user, final boolean removeCache) {
+        return remove(graphId, federatedAccess -> federatedAccess.hasWriteAccess(user), removeCache);
     }
 
-    protected boolean remove(final String graphId, final User user, final String adminAuth) {
-        return remove(graphId, federatedAccess -> federatedAccess.hasWriteAccess(user, adminAuth));
+    protected boolean remove(final String graphId, final User user, final boolean removeCache, final String adminAuth) {
+        return remove(graphId, federatedAccess -> federatedAccess.hasWriteAccess(user, adminAuth), removeCache);
     }
 
-    private boolean remove(final String graphId, final Predicate<FederatedAccess> accessPredicate) {
+    private boolean remove(final String graphId, final Predicate<FederatedAccess> accessPredicate, final boolean removeCache) {
         FederatedAccess accessFromCache = federatedStoreCache.getAccessFromCache(graphId);
         boolean rtn;
         if (nonNull(accessFromCache) && accessPredicate.test(accessFromCache)) {
+
+            removeGraphCaches(graphId, removeCache);
```

Comment:
I think this function should not have a boolean passed through to it, but instead should only be executed if `removeCache` is true. Effectively, remove the if statement from within the function and put it here